packages/@aws-cdk/aws-ec2/lib/launch-template.ts

```diff
@@ -0,0 +1,707 @@
+        // Will require creating an L2 for AWS::EC2::CapacityReservation
+        // capacityReservationSpecification: undefined,
+
+        // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata-cpuoptions.html
+        // cpuOptions: undefined,
+
+        // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-elasticgpuspecification.html
+        // elasticGpuSpecifications: undefined,
+
+        // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata.html#cfn-ec2-launchtemplate-launchtemplatedata-elasticinferenceaccelerators
+        // elasticInferenceAccelerators: undefined,
+
+        // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata.html#cfn-ec2-launchtemplate-launchtemplatedata-kernelid
+        // kernelId: undefined,
+        // ramDiskId: undefined,
+
+        // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata.html#cfn-ec2-launchtemplate-launchtemplatedata-licensespecifications
+        // Also not implemented in Instance L2
+        // licenseSpecifications: undefined,
+
+        // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata.html#cfn-ec2-launchtemplate-launchtemplatedata-metadataoptions
+        // metadataOptions: undefined,
+
+        // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata.html#cfn-ec2-launchtemplate-launchtemplatedata-tagspecifications
+        // Should be implemented via the Tagging aspect in CDK core. Complication will be that this tagging interface is very unique to LaunchTemplates.
+        // tagSpecification: undefined
+
+        // CDK has no abstraction for Network Interfaces yet.
+        // networkInterfaces: undefined,
+
+        // CDK has no abstraction for Placement yet.
+        // placement: undefined,
+
+      },
+    });
+
+    Tags.of(this).add(NAME_TAG, this.node.path);
+
+    this.defaultVersionNumber = resource.attrDefaultVersionNumber;
+    this.latestVersionNumber = resource.attrLatestVersionNumber;
+    this.launchTemplateId = resource.ref;
+    this.versionNumber = Token.asString(resource.getAtt('LatestVersionNumber'));
+  }
+
+  /**
+   * Allows specifying security group connections for the instance.
+   *
+   * @note Only available if you provide a securityGroup when constructing the LaunchTemplate.
+   */
+  public get connections(): Connections {
+    if (!this._connections) {
+      throw new Error('connections not available on LaunchTemplate. You must provide a securityGroup when constructing the LaunchTemplate to make it available.');
```

Comment:
Maybe phrase this in terms of APIs the end user would see (as opposed to APIs the author would see):

```
LaunchTemplate can only be used as IConnectable if a securityGroup is provided...
```